tests/types/test_uuid.py

```diff
@@ -32,7 +41,19 @@ def test_uuid_load(conn, fmt_out):
         copy.set_types(["uuid"])
         (res,) = copy.read_row()
 
-    assert res == UUID(val)
+    uuid_val = UUID(val)
+    assert res == uuid_val
+    # the C modules bypasses __init__, so checking the state of the UUID object
+    assert res.hex == uuid_val.hex
+    assert res.int == uuid_val.int
+    assert res.bytes == uuid_val.bytes
+    assert res.is_safe == uuid_val.is_safe
+    assert getattr(UUID, "__slots__") == (
+        "int",
+        "is_safe",
+        "__weakref__",
+    ), "UUID structure changed"
+    assert getattr(uuid_val, "__slots__") == getattr(UUID, "__slots__")
```

Comment:
Thank you for looking into the typeshed issue. I propose the following change.

```suggestion
    #  https://github.com/python/typeshed/issues/8832
    slots = ("int", "is_safe", "__weakref__")
    assert (
        UUID.__slots__ == slots  # type: ignore[attr-defined]
    ), "UUID structure changed"
```

I don't think we need to check the equality on the instances.